src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanEnterpriseBuildStep.java

```diff
@@ -545,7 +527,7 @@ public IResultsProvider call() throws AbortException {
 
 						while(m_scanStatus != null && (m_scanStatus.equalsIgnoreCase("Waiting to Auto Run") || m_scanStatus.equalsIgnoreCase("Waiting to Run")
 								|| m_scanStatus.equalsIgnoreCase("Starting") || m_scanStatus.equalsIgnoreCase("Running")
-								|| m_scanStatus.equals("Post Processing") || m_scanStatus.equals("Waiting to Generate Results") || m_scanStatus.equals("Generating Results"))) {
+								|| m_scanStatus.equals("Post Processing") || m_scanStatus.equals("Waiting to Generate Results") || m_scanStatus.equals("Generating Results") || m_scanStatus.equalsIgnoreCase("Canceling"))) {
```

Comment:
Given that there are now 8 different statuses being checked for to see if the scan is still executing, is there a smaller set that we could check for to indicate that the scan is completed?

Reply:
Sure, replacing the check with these conditions: "!(m_scanStatus.equalsIgnoreCase(CoreConstants.READY) || m_scanStatus.equalsIgnoreCase(CoreConstants.FAILED) || m_scanStatus.startsWith(CoreConstants.SUSPENDED)))"